tests/sentry/ingest/test_transaction_clusterer.py

```diff
@@ -310,3 +315,59 @@ def _get_projconfig_tx_rules(project: Project):
                 "redaction": {"method": "replace", "substitution": "*"},
             },
         ]
+
+
+@mock.patch("django.conf.settings.SENTRY_TRANSACTION_CLUSTERER_RUN", True)
+@mock.patch("sentry.ingest.transaction_clusterer.datasource.redis.MAX_SET_SIZE", 10)
+@mock.patch("sentry.ingest.transaction_clusterer.tasks.MERGE_THRESHOLD", 5)
+@mock.patch(
+    "sentry.ingest.transaction_clusterer.tasks.cluster_projects.delay",
+    wraps=cluster_projects,  # call immediately
+)
+@pytest.mark.django_db
+def test_transaction_clusterer_bumps_rules(_, default_organization):
+    tmp_redis_storage = {}
+
+    class MockRedisRuleStore:
+        def read(self, project: Project) -> RuleSet:
+            return tmp_redis_storage.get(project, {})
+
+        def write(self, project: Project, rules) -> None:
+            tmp_redis_storage[project] = rules
+
+    with mock.patch(
+        "sentry.ingest.transaction_clusterer.rules.RedisRuleStore", MockRedisRuleStore
+    ), Feature({"organizations:transaction-name-clusterer": True}):
+        project1 = Project(id=123, name="project1", organization_id=default_organization.id)
+        project1.save()
+        for i in range(10):
+            _store_transaction_name(project1, f"/user/tx-{project1.name}-{i}/settings")
+
+        with mock.patch("sentry.ingest.transaction_clusterer.rules._now", lambda: 1):
+            spawn_clusterers()
+
```

Comment:
Regarding the comment on "rogue" rules: Could you insert a `record_transaction_name` with `_meta.transaction.val: /i/am/rogue` on this line, to make sure that it does not appear in `_get_rules`?

Reply:
Added a separate test in [8394c18](https://github.com/getsentry/sentry/pull/46852/commits/8394c18a053185764b0ab6115611b970d71644a1).